apps/asap-server/test/mocks/algolia-client.mock.ts

```diff
@@ -1,10 +1,7 @@
-import { SearchClient, SearchIndex } from '@asap-hub/algolia';
+import { AlgoliaSearchClient } from '@asap-hub/algolia';
 
-export const algoliaIndexMock = {
-  saveObject: jest.fn(),
-  deleteObject: jest.fn(),
-} as unknown as jest.Mocked<SearchIndex>;
-
-export const algoliaClientMock = {
-  initIndex: jest.fn().mockReturnValue(algoliaIndexMock),
-} as unknown as jest.Mocked<SearchClient>;
+export const algoliaSearchClientMock = {
+  save: jest.fn(),
+  remove: jest.fn(),
+  searchEntity: jest.fn(),
+} as unknown as jest.Mocked<AlgoliaSearchClient>;
```

Comment:
We have to do the cast because of the constructor? Have we tried using the mocking built into jest?

Reply:
yes and no :D